test/unit/resources/aws_cloudwatch_alarm_test.rb

```diff
@@ -118,4 +118,44 @@ def setup
   def test_multiple_alarms_returned_fails
     assert_raises(RuntimeError) { AwsCloudwatchAlarm.new(metric_name: 'metric', metric_namespace: 'space', client_args: { stub_responses: true }, stub_data: [@data]) }
   end
+end
+
+class AwsCloudwatchAlarmWithDimensionsTest < Minitest::Test
```

Comment:
The new pattern for unit tests in inspec-aws seperate the actual tests and the mock data generation. For example, for the `aws_iam_user` resource, test data is generated in a [Mock class](https://github.com/inspec/inspec-aws/blob/master/test/unit/resources/mock/iam/aws_iam_user_mock.rb) which is subsequently used the the [actual test class](https://github.com/inspec/inspec-aws/blob/master/test/unit/resources/aws_iam_user_test.rb). It removes a lot of the boilerplate from the test classes and makes them easier to read and maintain.

They haven't all been updated to follow this pattern because there are so many, but it's good practice to update as you touch files. This would be a good opportunity for you to update this test class.

Reply:
I hadn't realised that this was the case. I'll take a look at `aws_iam_user ` and will update.

Reply:
After looking at this and speaking with @skpaterson I am going defer updating the tests.

Reply:
Why?